src/sentry/search/events/builder.py

```diff
@@ -1511,16 +1521,28 @@ def column(self, name: str) -> Column:
 
         :param name: The unresolved sentry name.
         """
+        missing_column = IncompatibleMetricsQuery(f"Column {name} was not found in metrics indexer")
+        if self.dry_run:
+            if name in DRY_RUN_COLUMNS:
+                return Column(name)
+            else:
+                raise missing_column
```

Comment:
I know we catch this and set tags, but just to guard against regressions, should we addd an endpoint test that purposely fails on dry run and make sure errors aren't returned?

Reply:
:+1: great idea